### PR TITLE
turn project into virtual workspace

### PR DESCRIPTION
**Pull request by reiner-dolp**
- clearer distinction between compiler library and cli interface
- more granular assignment of dependencies to crates
- more clear which tests run on which infrastructure (interation tests are part of compiler-cli, mj-tests are part of compiler-lib)
- allows for doc tests
- move asciifile back into compiler library
- move lexer tests in old main into lexer module

### CODE REVIEW

**Comment by flip1995** [comment]
1 commit - 355 files changed. Oh boi let's review :tada:

File: compiler-lib/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "hediet <>",
+    "joshuabach <>",
+    "problame <>",
+    "reiner-dolp <>"
+]
+edition = "2018"
+license = "MIT OR Apache-2.0"
+
+[dependencies]
+structopt = "0.2"
```
Comment (flip1995) [inline compiler-lib/Cargo.toml]: Do we need structopt in compiler-lib?

Comment (reiner-dolp) [reply compiler-lib/Cargo.toml]: Nope. :)

File: Cargo.toml [hunk 2]
```diff
@@ -1,34 +1,19 @@
...
-memmap = "0.7.0"
-libc = "*"
-termcolor = "1.0.4"
-derive_more = "0.13"
-asciifile = { path = "./asciifile" }
-strum = "0.11.0"
-strum_macros = "0.11.0"
+members = [
+    "compiler-lib",
+    "compiler-cli",
```
Comment (reiner-dolp) [inline Cargo.toml]: @flip1995 The only side-effect of listing crates seems to be a shared cargo.lock. Should we also list the integration test crates in compiler-cli?

Comment (flip1995) [reply Cargo.toml]: It would at least be more consistent to list them here, since we're also listing  `mjtest-rs/mjtest_macros` here.

Comment (reiner-dolp) [reply Cargo.toml]: or remove mjtest from the list

**Pull request merged**